### ISSUE TITLE
T5-M3-I2: Integrate Logging Into the File API

### ISSUE THREAD

**Issue by Dorkatron199** [rsanchez-wsu/jfiles#65]
# T5-M3-I2: Integrate Logging Into the File API

Continued from #36 (<b>T5-M2-I3: Integrate Logging Into the File API</b>), which wasn't done in the previous milestone...
## What

Team 5 just pulled log4j on 9 Oct. 2016 to team5-dev. This issue calls for team 5 to gain an understanding of log4j so that they can ensure that it is running in the File API. In other words, if it is not plug-and-play, what added work needs to be done so that log4j is ready in the File API?

Team 5 shall integrate the logging software made by team 2 into the JFile API. The team shall have at least one member who focuses on learning how to use it who the team can turn to. This member will work with the team members who are working on specific classes to get the logging software implemented in those classes. Since JFile is mostly finished, that class should be done first.
## Why

This will be necessary for both troubleshooting, especially once the API is implemented in the JFiles program, and documentation. It is also required to meet the standards of the rest of the teams. Additionally, having one member who knows the logging software very well will undoubtedly prevent many pitfalls in the future.
## How

Use the built-in java documentation and the wiki devoted to this subject.

**Comment by Dorkatron199** [comment]
Briefly looking into log4j, I can see that it does require coding.

Sergio has my permission to add anything and everything necessary to JFileManager to ensure that it logs appropriately, once it gets far enough along to need it. I am sure Brand will also give permission. If there's nothing for you do do in this issue, you can create another issue for making the documentation for JFile more thorough.

This will be useful 1) so that you can have a chance to commit to the project, 2) so that you get a chance to review JFile a bit and get a feel for what it does, and 3) enures that there is a second pair of eyes reviewing Brand's code more thoroughly. I did some brief once-overs, but this will help to effectively peer-review. This will also be done to my code in JFileManager once it is ready.

**Comment by srodriguez4** [comment]
**Added Log4j Methods to JFIle and JFileManager**
Added methods:
- logger.info("insert message"); To log a certain functionality that has
  been executed and log that it has been successfully executed upon
  completion.
- logger.error("insert message"); To log in the event an error has taken
  place after a certain command has taken place.

These logging methods still need to be implemented appropriately within the functionality of the methods that are executing delete, copy, paste, etc. within try-catch blocks. They need to be in try-catch blocks to indicate and log that delete, copy, paste, etc. has been triggered, successfully completed, or log that an error has occurred.

**Comment by rsanchez-wsu** [comment]
@srodriguez4 I am confused by your last comment and the closure of this issue.  When you say "added methods" are those added to documentation, to specification, or to code?  Is there a commit associated with this?

**Comment by Dorkatron199** [comment]
This was done via commit ce852a9.

Since part of his task was to learn what methods need to be implemented from log4j for us to be in compliance with the rest of the project (and in the context of the things he changed in the commit), I think this may be information for us. He asked me if it was okay for him to close this issue and I gave him permission.

By "Added methods:" he means he implemented these methods inside of our to start recording logs. Documentation and specifications have not been updated yet. Additionally, the methods are not in the try-catch blocks (where they will record the appropriate information), they are just thrown in.

Ideally, we will create try-catch blocks for each method and put the "start" log before the try, the "complete" log at the end of the try, and the "error" in the catch block.

This isn't my issue, so I don't have all of the issues, but it isn't showing any compile errors in the editor. I also still don't have a working Ant on my system, so either Sergio or Brand will have to check. I apologize for my late reply.

**Comment by Dorkatron199** [comment]
His code has now been consolidated into try/catch blocks. What he did _was_ enough to close the issue, but I went ahead and reworked the code to clear out redundancies and make the logger log correctly.

**Comment by Dorkatron199** [comment]
Sergio typed "ref# 65" by accident when doing the commit.

**Comment by rsanchez-wsu** [comment]
@Dorkatron199 Thanks for the clarification
